### PR TITLE
Inspec beta flag on resources

### PR DESCRIPTION
**Pull request by slevenick**
Generate InSpec resources at beta.

Switch on endpoint due to flag given in resource constructor. No flag means use GA endpoint, `:beta = true` means use the beta endpoint

**Release Note Template for Downstream PRs (will be copied)**

```release-note:REPLACEME

```

### CODE REVIEW

File: provider/inspec.rb [hunk 1]
```diff
@@ -265,7 +265,9 @@ def resource_name(object, product)
     # Recursively calls itself on any arrays or nested objects within this property, indenting
     # further for each call
     def markdown_format(property, indent = 1)
-      prop_description = "`#{property.out_name}`: #{property.description.split("\n").join(' ')}"
+      beta_description = property.min_version.name == 'beta' ? '(Beta only) ' : ''
```
Comment (rambleraptor) [inline provider/inspec.rb]: Is min_version always going to be not nil?

File: templates/inspec/singular_resource.erb [hunk 2]
```diff
@@ -42,14 +42,14 @@ class <%= object.__product.name.camelize(:upper) -%><%= object.name -%> < GcpRes
   def initialize(params)
     super(params.merge({ use_http_transport: true }))
     @params = params
-    @fetched = @connection.fetch(product_url, resource_base_url, params, 'Get')
+    @fetched = @connection.fetch(product_url(params[:beta]), resource_base_url, params, 'Get')
```
Comment (rambleraptor) [inline templates/inspec/singular_resource.erb]: What's the purpose of params[:beta]? Just a way for users to explicitly set if they want to use the beta APIs or not?

Comment (slevenick) [reply templates/inspec/singular_resource.erb]: Exactly

File: provider/inspec.rb (hunk 1 above)
Comment (slevenick) [reply provider/inspec.rb]: Correct: https://github.com/GoogleCloudPlatform/magic-modules/blob/1307550a3ab01b20c2f8410d51c191e190d33d30/api/type.rb#L249

**Pull request merged**